atom/feature_engineering.py

```diff
@@ -988,12 +1014,68 @@ def check_y():
                 **self._kwargs,
             ).fit(X, y)
 
+        elif self.strategy.lower() in ["pso", "gwo", "dfo", "geneo", "hho"]:
+            check_y()
+            algo_mapper = {"pso": ParticleSwarmOptimization, # mapper to avoide code repetition
+                           "gwo": GreyWolfOptimization,
+                           "dfo": DragonFlyOptimization,
+                           "geneo": GeneticOptimization,
+                           "hho": HarrisHawkOptimization}
+            
+            # initialization_params catches all params requied for zoofs algos
+            initialization_params = {"pso": ['n_iteration', 'timeout', 'population_size', 'minimize', 'c1', 'c2', 'w'],
+                                     "gwo": ['n_iteration', 'timeout', 'population_size', 'minimize', 'method'],
+                                     "dfo": ['n_iteration', 'timeout', 'population_size', 'minimize', 'method'],
+                                     "geneo": ['n_iteration', 'timeout', 'population_size', 'minimize', 'selective_pressure',
+                                               'elitism', 'mutation_rate'],
+                                     "hho": ['n_iteration', 'timeout', 'population_size', 'minimize', 'beta']}[self.strategy.lower()]
+            initialization_params_from_kwargs = {
+                k: v for k, v in self.kwargs.items() if k in initialization_params}
+            params_for_objective_function = {k: v for k, v in self.kwargs.items(
+            ) if k not in initialization_params+['objective_function']}
+
+            
+            if self.kwargs.get("objective_function"): # zoofs-native objective_function choice
+                objective_function = self.kwargs['objective_function']
+
+            elif self.kwargs.get("scoring"): # atom-native scoring method can be chosen 
+                params_for_objective_function = {
+                    "scorer": get_custom_scorer(self.kwargs["scoring"])}
+                objective_function = custom_function_for_scorer
+
+            else: # default scoring method is chosen if nothing is provided
+                params_for_objective_function = {"scorer": get_custom_scorer("neg_log_loss") if hasattr(
```

Comment:
atom uses default scorers `f1` for binary classification, `f1_weighted` for multiclass classification and `r2` for regression. You can check which one it is with the attribute `goal`. I think it's best to change them here to atom's default values for consistency